.github/workflows/deploy_cms.yaml

```diff
@@ -0,0 +1,74 @@
+name: Deploy CMS
+
+on:
+  push:
+        branches: 'main'
+        # Run only if there are at least one change matching the following paths
+        paths:
+          - 'apps/strapi-cms/*'
+          - '.github/workflows/deploy_cms.yaml'
+
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+        type: choice
+        options:
+        - info
+        - warning
+        - debug
+      environment:
+        description: 'The environment used as target'
+        type: choice
+        required: true
+        default: prod
+        options:
+        - prod
+
+defaults:
+  run:
+    shell: bash
+    working-directory: apps/strapi-cms
+
+permissions:
+  id-token: write   # This is required for requesting the JWT
+  contents: read    # This is required for actions/checkout
+
+jobs:
+  deploy:
+    name: Docker CMS
+    runs-on: ubuntu-latest
+    continue-on-error: false
+    environment: ${{ inputs.environment }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+    
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3.0.0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+
+      - name: Configure AWS Credentials
+        uses: ./.github/actions/configure-aws-credentials
+        with:
+          aws_region: ${{ env.AWS_REGION || 'eu-south-1' }}
+          role_to_assume: ${{ secrets.IAM_ROLE }}
+
+      - name: Docker image Build
+        run: docker build -t cms-strapi:latest .
+
+      - name: Docker image Tag
+        run: docker tag cms-strapi:latest 949355639119.dkr.ecr.eu-south-1.amazonaws.com/cms-strapi:latest
+
+      - name: Login to ECR
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+        with:
+            registry: 949355639119.dkr.ecr.eu-south-1.amazonaws.com
```

Comment:
Just a question here: should we treat the registry URL as a sensitive information and get it from the environment?

Reply:
Yes, we can use ${{ steps.login-ecr.outputs.registry }} for reference to url registry